frontend/pkg/frontend/frontend.go

```diff
@@ -86,11 +86,21 @@ func (f *Frontend) Run(ctx context.Context, stop <-chan struct{}) {
 	// This just digs up the logger passed to NewFrontend.
 	logger := LoggerFromContext(f.server.BaseContext(f.listener))
 
+	errs, ctx := errgroup.WithContext(ctx)
+	errs.Go(func() error {
+		shutdown, err := InstallOpenTelemetryTracer(ctx, logger)
+		defer func() {
+			if err := shutdown(ctx); err != nil {
+				logger.Error("otel-sdk shutdown failed", "error", err)
+			}
+		}()
```

Comment:
Looks like this defer function should be conditional.  If `InstallOpenTelemetryTracer` fails then the `shutdown` function will be `nil`, and attempting to call it will panic.  Am I reading that right?